include/range/v3/range/access.hpp

```diff
@@ -107,16 +115,16 @@ namespace ranges
 #ifdef RANGES_WORKAROUND_GCC_89953
             constexpr auto CPP_fun(operator())(R && r)(
                 const                                   //
-                noexcept(noexcept(impl_v<R>((R &&) r))) //
-                requires(has_member_begin<R> || has_non_member_begin<R>))
+                noexcept(noexcept(impl_v<R>(r))) //
+                requires(_safe_::safe<R> && (has_member_begin<R> || has_non_member_begin<R>)))
 #else
             constexpr auto CPP_fun(operator())(R && r)(
                 const                                   //
-                noexcept(noexcept(impl<R>{}((R &&) r))) //
-                requires(has_member_begin<R> || has_non_member_begin<R>))
+                noexcept(noexcept(impl<R>{}(r))) //
+                requires(_safe_::safe<R> && (has_member_begin<R> || has_non_member_begin<R>)))
 #endif
             {
-                return impl<R>{}((R &&) r);
+                return impl<R>{}(r);
             }
 
 #if defined(__cpp_lib_string_view) && __cpp_lib_string_view > 0
```

Comment:
Perhaps it would be better now to implement this as a partial specialization of `enable_safe_view` on `basic_string_view` instead of this ugly hack in the `begin` CPO.